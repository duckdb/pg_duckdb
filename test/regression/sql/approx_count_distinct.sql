CREATE TABLE t (a int, b text);
INSERT INTO t VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd'), (5, 'e');
INSERT INTO t VALUES (2, 'f'), (3, 'g'), (4, 'h');
SELECT approx_count_distinct(a), approx_count_distinct(b) FROM t;
SELECT a, approx_count_distinct(b) FROM t GROUP BY a ORDER BY a;
SELECT a, approx_count_distinct(b) OVER (PARTITION BY a) FROM t ORDER BY a;
DROP TABLE t;
