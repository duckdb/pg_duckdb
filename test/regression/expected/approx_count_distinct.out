CREATE TABLE t (a int, b text);
INSERT INTO t VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd'), (5, 'e');
INSERT INTO t VALUES (2, 'f'), (3, 'g'), (4, 'h');
SELECT approx_count_distinct(a), approx_count_distinct(b) FROM t;
 approx_count_distinct | approx_count_distinct 
-----------------------+-----------------------
                     5 |                     9
(1 row)

SELECT a, approx_count_distinct(b) FROM t GROUP BY a ORDER BY a;
 a | approx_count_distinct 
---+-----------------------
 1 |                     1
 2 |                     2
 3 |                     2
 4 |                     2
 5 |                     1
(5 rows)

SELECT a, approx_count_distinct(b) OVER (PARTITION BY a) FROM t ORDER BY a;
 a | approx_count_distinct 
---+-----------------------
 1 |                     1
 2 |                     2
 2 |                     2
 3 |                     2
 3 |                     2
 4 |                     2
 4 |                     2
 5 |                     1
(8 rows)

DROP TABLE t;
